riscv_private.h

```diff
@@ -333,6 +333,17 @@ static inline uint32_t calc_fclass(uint32_t f) {
 
   return out;
 }
+static inline bool is_nan(uint32_t f){
+  const uint32_t expn = f & FMASK_EXPN;
+  const uint32_t frac = f & FMASK_FRAC;
+  return (expn == FMASK_EXPN && frac);
+}
+
+static inline bool is_signan(uint32_t f){
```

Comment:
In [Floating-point exceptions](https://www.ibm.com/docs/en/aix/7.3?topic=concepts-floating-point-exceptions) of IBM AIX Manual, signaling NaN is abbreviated as `snan`, which might represent better than "signan." In Linux, the prefix "sig" is used for UNIX signal specific identifiers. However, signaling NaN is quite different from them. Let's name it `is_snan` here.